bitcoin/src/taproot/mod.rs

```diff
@@ -1180,6 +1181,12 @@ impl ControlBlock {
 
         Ok(ControlBlock { leaf_version, output_key_parity, internal_key, merkle_branch })
     }
+
+    /// Constructs a new [`ControlBlock`] from a hex string.
+    pub fn from_hex(hex: &str) -> Result<Self, TaprootError> {
+        let vec = Vec::from_hex(hex).map_err(TaprootError::InvalidControlBlockHex)?;
+        ControlBlock::decode(vec.as_slice())
+    }
```

Comment:
I'm thinking of cleaning up the error types but I guess this doesn't need to be blocked on that.

Reply:
@Kixunil you may be interested in my Miniscript taproot work -- in particular https://github.com/apoelstra/rust-miniscript/blob/2025-03--taproot-api/src/descriptor/tr/spend_info.rs and https://github.com/apoelstra/rust-miniscript/blob/2025-03--taproot-api/src/descriptor/tr/taptree.rs

I'd kinda like to replace our `SpendInfo` structure here with something like this, though I'm not sure what that'd look like if I tried to minimize the duplicated code between rust-bitcoin and rust-miniscript. Maybe there's just no way to do it since rust-miniscript has generic keys.

Anyway I think it's fine to hold off on any error cleanups until we think about that too.

Reply:
I'm also thinking that we should only need one length error variant here - there the set of all valid input lengths can be described as `2 * (33 + 32 * n)` where `n` is an integer in range `0..=128`, so we can just have it carry a single `usize`.